### PR TITLE
backfill add score process

### LINKED ISSUE #2941: The task-topology plugin cannot work with the tasks with empty resource request
**What happened**:

Assume that we have a kubernetes cluster with two nodes and a simple job with topology annotations:

```yaml
apiVersion: batch.volcano.sh/v1alpha1
kind: Job
metadata:
  annotations:
    volcano.sh/task-topology-anti-affinity: "nginx"
  name: example-job-1
spec:
  minAvailable: 5
  schedulerName: volcano
  plugins:
    ssh: []
    svc: []
  tasks:
    - replicas: 2
      name: nginx
      template:
        spec:
          containers:
            - image: nginx
              name: nginx-main
          restartPolicy: OnFailure
    - replicas: 3
      name: mysql
      template:
        spec:
          containers:
            - env:
                - name: MYSQL_ROOT_PASSWORD
                  value: "123456"
              image: mysql
              name: mysql-main
          restartPolicy: OnFailure
```

And the scheduler config file:

```yaml
actions: "enqueue, backfill"
tiers:
  - plugins:
      - name: priority
      - name: gang
        enablePreemptable: false
      - name: task-topology
```

We create two kind of tasks and we want the two nginx tasks are scheduled into different nodes using the task topology antiAffinity annotations.

But the two nginx tasks are scheduled into the same node in most time:

image

If we add resource requests to the job spec, all things work well as expected:

```yaml
apiVersion: batch.volcano.sh/v1alpha1
kind: Job
metadata:
  annotations:
    volcano.sh/task-topology-anti-affinity: "nginx"
  name: example-job-1
spec:
  minAvailable: 5
  schedulerName: volcano
  plugins:
    ssh: []
    svc: []
  tasks:
    - replicas: 2
      name: nginx
      template:
        spec:
          containers:
            - image: nginx
              name: nginx-main
              resources:
                requests:
                  cpu: "0.1"
          restartPolicy: OnFailure
    - replicas: 3
      name: mysql
      template:
        spec:
          containers:
            - env:
                - name: MYSQL_ROOT_PASSWORD
                  value: "123456"
              image: mysql
              name: mysql-main
              resources:
                requests:
                  cpu: "0.3"
          restartPolicy: OnFailure
```

image

**What you expected to happen**:

The task-topology plugin should work no matter whether the tasks have resource requests.

**How to reproduce it (as minimally and precisely as possible)**:

As described above.

**Anything else we need to know?**:

After reading the code, I found that the allocate action will be skipped if the task has empty resource request: https://github.com/volcano-sh/volcano/blob/ed5c215d415d98845949d0fde6707cab29621989/pkg/scheduler/actions/allocate/allocate.go#L135-L139

This will cause the `TaskOrderFn` and `NodeOrderFn` in `topology.go` be not executed, which may cause the tasks are scheduled into incorrect nodes.

**Environment**:
- Volcano Version: v1.7.0
- Kubernetes version (use `kubectl version`):
  Client Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.2", GitCommit:"7f6f68fdabc4df88cfea2dcf9a19b2b830f1e647", GitTreeState:"clean", BuildDate:"2023-05-17T14:20:07Z", GoVersion:"go1.20.4", Compiler:"gc", Platform:"darwin/amd64"}
  Kustomize Version: v5.0.1
  Server Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.3", GitCommit:"9e644106593f3f4aa98f8a84b23db5fa378900bd", GitTreeState:"clean", BuildDate:"2023-03-15T13:33:12Z", GoVersion:"go1.19.7", Compiler:"gc", Platform:"linux/amd64"}
- Cloud provider or hardware configuration: minikube using hyperkit driver on macOS(Intel)
- Kernel (e.g. `uname -a`): `Linux volcano-demo 5.10.57 #1 SMP Mon Apr 3 23:35:10 UTC 2023 x86_64 GNU/Linux`

### PR DESCRIPTION
**Pull request by lowang-bh**
Fix #2941 add priority score in backfill actions

image

>
Note:
  I have try to add ut, but predicate and nodeorder plugin relay on k8s client which can not be overwrite with fakeclient.
Please merge PR https://github.com/volcano-sh/volcano/pull/2866 with high priority and then can use interface to overwite k8sclient

### CODE REVIEW

**Comment by lowang-bh** [comment]
@hwdef @wangyang0616 @Thor-wl Would please help to review it?

CC @william-wang @k82cn

File: pkg/scheduler/actions/backfill/backfill.go [hunk 1]
```diff
@@ -44,6 +44,21 @@ func (backfill *Action) Execute(ssn *framework.Session) {
...
 	defer klog.V(5).Infof("Leaving Backfill ...")
 
+	predicatFunc := func(task *api.TaskInfo, node *api.NodeInfo) ([]*api.Status, error) {
+		var statusSets util.StatusSets
+		statusSets, err := ssn.PredicateFn(task, node)
+		if err != nil {
+			return nil, err
+		}
+
+		// predicateHelper.PredicateNodes will print the log if predicate failed
```
Comment (hwdef) [inline pkg/scheduler/actions/backfill/backfill.go:54]: What is the use of this comment?

File: pkg/scheduler/actions/backfill/backfill.go [hunk 2]
```diff
@@ -69,42 +86,33 @@ func (backfill *Action) Execute(ssn *framework.Session) {
...
+						node = util.SelectBestNode(nodeScores)
 					}
-
-					metrics.UpdateE2eSchedulingDurationByJob(job.Name, string(job.Queue), job.Namespace, metrics.Duration(job.CreationTimestamp.Time))
-					metrics.UpdateE2eSchedulingLastTimeByJob(job.Name, string(job.Queue), job.Namespace, time.Now())
-					allocated = true
-					break
 				}
+				// As task did not request resources, so it only need to meet predicates.
+				// Done (k82cn): need to prioritize nodes to avoid pod hole.
```
Comment (hwdef) [inline pkg/scheduler/actions/backfill/backfill.go:105]: This can also be deleted

**Review by hwdef — COMMENTED** [review]
I think this is great!
Thanks for your contributions!

File: pkg/scheduler/actions/backfill/backfill.go (hunk 2 above)
Comment (lowang-bh) [reply pkg/scheduler/actions/backfill/backfill.go]: Done.

File: pkg/scheduler/actions/backfill/backfill.go (hunk 1 above)
Comment (lowang-bh) [reply pkg/scheduler/actions/backfill/backfill.go]: To explain why don't pring log here.

**Review by hwdef — COMMENTED** [review]
/lgtm

**Comment by lowang-bh** [comment]
Hi, @k82cn could you please help to review? Thanks.

**Review by william-wang — APPROVED** [review]
/approve

**Pull request merged**